BoggleBoard.java

```diff
@@ -0,0 +1,293 @@
+        soln[111] = "ream";
+        soln[112] = "reap";
+        soln[113] = "reede";
+        soln[114] = "rein";
+        soln[115] = "rended";
+        soln[116] = "rented";
+        soln[117] = "reteam";
+        soln[118] = "riched";
+        soln[119] = "rinded";
+        soln[120] = "taed";
+        soln[121] = "tape";
+        soln[122] = "teaed";
+        soln[123] = "teed";
+        soln[124] = "teen";
+        soln[125] = "teener";
+        soln[126] = "teind";
+        soln[127] = "tench";
+        soln[128] = "tene";
+        soln[129] = "terne";
+        soln[130] = "apted";
+        soln[131] = "cheer";
+        soln[132] = "chin";
+        soln[133] = "chiro";
+        soln[134] = "cire";
+        soln[135] = "deen";
+        soln[136] = "deet";
+        soln[137] = "deni";
+        soln[138] = "dented";
+        soln[139] = "eater";
+        soln[140] = "eine";
+        soln[141] = "enriched";
+        soln[142] = "enter";
+        soln[143] = "erne";
+        soln[144] = "eten";
+        soln[145] = "held";
+        soln[146] = "hented";
+        soln[147] = "hetero";
+        soln[148] = "hinted";
+        soln[149] = "hiree";
+        soln[150] = "inched";
+        soln[151] = "inro";
+        soln[152] = "leed";
+        soln[153] = "leet";
+        soln[154] = "mate";
+        soln[155] = "meat";
+        soln[156] = "medle";
+        soln[157] = "meet";
+        soln[158] = "mete";
+        soln[159] = "meteoric";
+        soln[160] = "neap";
+        soln[161] = "neeld";
+        soln[162] = "nete";
+        soln[163] = "open";
+        soln[164] = "opter";
+        soln[165] = "pate";
+        soln[166] = "pater";
+        soln[167] = "peel";
+        soln[168] = "peined";
+        soln[169] = "pene";
+        soln[170] = "pent";
+        soln[171] = "pern";
+        soln[172] = "porn";
+        soln[173] = "preen";
+        soln[174] = "preteen";
+        soln[175] = "printed";
+        soln[176] = "reame";
+        soln[177] = "reate";
+        soln[178] = "reel";
+        soln[179] = "reined";
+        soln[180] = "rent";
+        soln[181] = "repo";
+        soln[182] = "retem";
+        soln[183] = "richen";
+        soln[184] = "rine";
+        soln[185] = "tame";
+        soln[186] = "tapen";
+        soln[187] = "team";
+        soln[188] = "teel";
+        soln[189] = "teend";
+        soln[190] = "teenier";
+        soln[191] = "teinded";
+        soln[192] = "tend";
+        soln[193] = "tepa";
+        soln[194] = "terned";
+    
+    
+        for(int i = 0; i < h; i++){
+            for(int j = 0; j < w; j++){
+               boggle(myBoard, visited, word, soln, i, j);
+            }
+        }
+    }
+    
+    private static void boggle(char[][] board, char[][] visited, String word, String[]soln, int x, int y){
+        
+        if(visited[x][y] == '1')
+            return;
+        
+        visited[x][y] = '1';
+        word += board[x][y];
+        
+        if(Arrays.asList(soln).contains(word))
```

Comment:
This will repeatedly create a lot of lists from the same array, many many times.